src/app/components/fees/fee-item-icon.tsx

```diff
@@ -0,0 +1,23 @@
+import {
+  AnimalChameleonIcon,
+  AnimalEagleIcon,
+  AnimalRabbitIcon,
+  AnimalSnailIcon,
+} from '@leather.io/ui';
+
+import type { FeeType } from '@app/common/fees/use-fees';
+
+import { IconWrapper } from '../icon-wrapper';
+
+export function FeeItemIcon({ feeType }: { feeType: FeeType }) {
+  function getIcon() {
+    if (feeType === 'slow') return <AnimalSnailIcon variant="small" />;
+    if (feeType === 'standard') return <AnimalRabbitIcon variant="small" />;
+    if (feeType === 'fast') return <AnimalEagleIcon variant="small" />;
+    if (feeType === 'custom') return <AnimalChameleonIcon variant="small" />;
+
+    throw new Error('Invalid fee type');
+  }
```

Comment:
Correct me if mistaken, but in React this is bad for perf. as it can't do any caching of the JSX and rerenders it every time. 

Better written as:


```tsx
const feeTypeToIconMap: Record<FeeType, ReactNode> = {
  slow: <AnimalSnailIcon variant="small" />,
  standard: <AnimalRabbitIcon variant="small" />,
  fast: <AnimalEagleIcon variant="small" />,
  custom: <AnimalChameleonIcon variant="small" />,
};

export function FeeItemIcon({ feeType }: { feeType: FeeType }) {
  const icon = feeTypeToIconMap[feeType] || null;

  if (!icon) {
    throw new Error('Invalid fee type');
  }

  return <IconWrapper>{icon}</IconWrapper>;
}
```

This example isn't even optimal, as we render all 4 instead of just one, but still more performant than everytime afaik